src/as_prover.mli

```diff
@@ -29,6 +29,18 @@ module type Basic = sig
     -> 'var
     -> ('value, 'f field, 'prover_state) t
 
+  module Provider : sig
```

Comment:
Is there a reason you moved this into this module? Some kind of dependency or something?

Reply:
It used to depend on the specific `As_prover0.t` that we used as the canonical `As_prover.t`. To functor over the `As_prover` interface, it also needs to depend on the specific instance of `As_prover`, and so this seemed like a logical place to put it because of its dependency on the `As_prover` runner.